subsys/dfu/boot/mcuboot.c

```diff
@@ -616,7 +622,7 @@ int mcuboot_swap_type(void)
 		return rc;
 	}
 
-	rc = boot_read_swap_state_by_id(FLASH_AREA_IMAGE_SECONDARY,
+	rc = boot_read_swap_state_by_id(UPLOAD_FLASH_AREA_ID,
```

Comment:
I am not sure this is a good idea, the tables were not build with comparing a partition against itself in mind. I would just check if `UPLOAD_FLASH_AREA_ID` equals `FLASH_AREA_IMAGE_PRIMARY` at the beginning and return `BOOT_SWAP_TYPE_NONE`